src/modules/app/StateType.js

```diff
@@ -2,49 +2,38 @@
 
 import type { StoreActionType } from './StoreActionType'
 import type { PersistState } from 'redux-persist/src/types'
-import { CategoryModel, CityModel, LanguageModel } from '@integreat-app/integreat-api-client'
+import { CategoryModel, CityModel, EventModel, LanguageModel } from '@integreat-app/integreat-api-client'
 import type { ResourceCacheType } from '../endpoint/ResourceCacheType'
 
-export type LanguageStateType = string | null
-
-export type CurrentCityStateType = string | null
-
 type PathType = string
 
-export type RouteStateType = {
-  +root: ?string,
+export type CategoryRouteStateType = {|
+  +root: ?string, // path of the root category
   +depth: number,
-  +models: { [path: PathType]: CategoryModel }, /* Models could be stored outside of RouteStateType
+  +models: { [path: PathType]: CategoryModel }, /* Models could be stored outside of CategoryRouteStateType
                                                    (e.g. CategoriesStateType) to save memory
                                                    in the state. This would be an optimization! */
   +children: { [path: PathType]: Array<PathType> }
-}
+|}
+
+export type EventRouteStateType = {|
+  +path: string | null,
+  +models: Array<EventModel>
+|}
 
-export const defaultRouteState: RouteStateType = {
+export const defaultRouteState: CategoryRouteStateType = {
   root: null,
   models: {},
   children: {},
   depth: 0
 }
 
-export type CategoriesStateType = {
-  +routeMapping: {
-    [key: string]: RouteStateType
-  },
-  resourceCache: ResourceCacheType,
-
-  +languages: Array<LanguageModel>,
-  +currentLanguage: LanguageStateType,
-  +currentCity: CurrentCityStateType
+export type CategoriesRouteMappingType = {
```

Comment:
I guess we should also use `{| ...|}` here and for the EventsRouteMappingType.

Reply:
Exact objects do not make sense if the object is of arbitrary length